browser/brave_content_browser_client.cc

```diff
@@ -1250,3 +1243,23 @@ blink::UserAgentMetadata BraveContentBrowserClient::GetUserAgentMetadata() {
       base::StrCat({base::NumberToString(version.components()[0]), ".0.0.0"});
   return metadata;
 }
+
+content::StoragePartitionConfig
+BraveContentBrowserClient::GetStoragePartitionConfigForSite(
+    content::BrowserContext* browser_context,
+    const GURL& site) {
+  if (auto* request_otr_service =
+          request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+              browser_context)) {
+    if (request_otr_service->RequestedOTR(site)) {
+      CHECK(site.has_host());  // upstream also does this before accessing
+                               // site.host()
+      return content::StoragePartitionConfig::Create(
+          browser_context, site.host(), /*partition_name=*/std::string(),
```

Comment:
maybe we should give the partition a useful name like `"request_otr"`?

Reply:
named in https://github.com/brave/brave-core/commit/c83ae23838b8132c40d9ce007c338abf52804ae9